app/src/main/java/com/github/se/travelpouch/ui/home/TravelList.kt

```diff
@@ -209,3 +262,112 @@ fun TravelItem(travelContainer: TravelContainer, onClick: () -> Unit) {
+
+  Column(modifier = Modifier.fillMaxWidth()) {
+    Box(modifier = Modifier.fillMaxWidth().requiredHeight(mapHeight.floatValue.dp)) {
+      MapContent(
+          modifier = Modifier.fillMaxWidth().height(mapHeight.floatValue.dp),
+          travelContainers = travelList.value,
+          onInfoWindowClickCallback = { travelContainer ->
+            // Handle the click event
+            onInfoWindowClickCallback(travelContainer)
+          })
+    }
+
+    // Strap handle at the bottom to drag and resize the map
+    Box(
+        modifier =
+            Modifier.fillMaxWidth()
+                .testTag("mapLatch")
+                .graphicsLayer {
+                  shape =
+                      CutCornerShape(
+                          topStart = 0.dp, topEnd = 0.dp, bottomStart = 16.dp, bottomEnd = 16.dp)
+                  clip = true
+                }
+                .height(latchDp)
+                .background(MaterialTheme.colorScheme.tertiary)
+                .draggable(
+                    orientation = Orientation.Vertical,
+                    state =
+                        rememberDraggableState { delta ->
+                          resizeFromDragMotion(
+                              delta, density, mapHeight, maxHeightPx, minHeightPx, isCollapsed)
+                        })) {
+          Icon(
+              imageVector = Icons.Default.StopCircle,
+              contentDescription = "MapLatch",
+              modifier = Modifier.align(Alignment.Center))
+        }
+  }
+}
+
+/**
+ * Adjusts the height of the map based on the drag motion.
+ *
+ * @param delta The change in position from the drag motion.
+ * @param density The density scale factor.
+ * @param mapHeight The current height of the map.
+ * @param maxHeightPx The maximum height of the map in pixels.
+ * @param minHeightPx The minimum height of the map in pixels.
+ * @param isCollapsed A state indicating whether the map is collapsed.
+ */
+private fun resizeFromDragMotion(
+    delta: Float,
+    density: Density,
+    mapHeight: MutableFloatState,
+    maxHeightPx: Float,
+    minHeightPx: Float,
+    isCollapsed: MutableState<Boolean>
+) {
+  val scaledDelta = delta / density.density // Adjusting by the density scale factor
+
+  // Calculate the new map height based on the delta drag amount
+  // clamp to the max value to avoid exceeding the max height
+  if (mapHeight.floatValue + scaledDelta > maxHeightPx) {
+    mapHeight.floatValue = maxHeightPx
+  } else if (mapHeight.floatValue + scaledDelta < minHeightPx && !isCollapsed.value) {
+    isCollapsed.value = true
+    mapHeight.floatValue = 0f
+  } else {
+    if (isCollapsed.value && mapHeight.floatValue + scaledDelta > minHeightPx) {
+      isCollapsed.value = false
+    }
+    // clamp to the min value to avoid negatives values
+    if (mapHeight.floatValue + scaledDelta < 0) {
+      mapHeight.floatValue = 0f
+    } else {
+      mapHeight.floatValue += scaledDelta
+    }
+  }
+}
```

Comment:
I think these are lot of conditions. Can't you just reduce it to 3 conditions : 
1) mapHeight + scaled delta > max ==> mapHeight = max and isCollapsed = false
2) mapHeight + scaled delta > min ==> mapHeight = 0 and isCollapsed = true
3) else : mapHeight += scaled delta and isCollapsed = false

Since setting isCollapsed is very cheap, even neglectable.